### PR TITLE
[Telestrat] chdir is now working

### PR DESCRIPTION
**Pull request by jedeoric**
add chdir management

### CODE REVIEW

**Comment by jedeoric** [comment]
There is a problem with this PR because i did not update initcwd. I try to fix it (i wanted to do a separate PR, but it's not clear if i split into 2 PR)

**Comment by polluks** [comment]
@oliverschmidt How about some more .gitattributes magic?
https://stackoverflow.com/questions/13819764/indentation-configuration-only-for-some-files/57244091#57244091

**Comment by jedeoric** [comment]
I forgot to say that i finished all corrections.

File: libsrc/telestrat/syschdir.s [hunk 1]
```diff
@@ -0,0 +1,34 @@
+;
+; Jede (jede@oric.org), 2021-02-22
+;
+; unsigned char _syschdir (const char* name, ...);
+;
+
+        .export         __syschdir
+        .import         addysp, popax
+	    .importzp       tmp1
```
Comment (oliverschmidt) [inline libsrc/telestrat/syschdir.s:9]: Undesired tab char.

File: libsrc/telestrat/syschdir.s [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        stx     tmp1
+        ldy     tmp1
+        
+        ; Call telemon primitive
+        
+        BRK_TELEMON(XPUTCWD)
+
+        jsr     initcwd      ; Update cwd
+        
+        rts
```
Comment (oliverschmidt) [inline libsrc/telestrat/syschdir.s:34]: Tail call optimization opportunity: `jsr <xxx> rts` -> `jmp <xxx>`.

File: libsrc/telestrat/syschdir.s (hunk 1 above)
Comment (jedeoric) [reply libsrc/telestrat/syschdir.s]: Fixed

File: libsrc/telestrat/syschdir.s (hunk 2 above)
Comment (jedeoric) [reply libsrc/telestrat/syschdir.s]: Fixed

**Comment by oliverschmidt** [comment]
I had to revert this pull request as it doesn't even build :-(

**Comment by greg-king5** [comment]
@jedeoric,
Definitions for `XGETCWD` and `XPUTCWD` need to be added to "asminc/telestrat.inc".

**Comment by jedeoric** [comment]
@greg-king5 Sorry for the mistake, because i did not pushed the file. That is why it was working here, but not in this PR.

Anyway, i think that i have to open a new PR ? I doubt that i can open again this PR.

**Pull request merged**